great_expectations/execution_engine/execution_engine.py

```diff
@@ -277,22 +281,57 @@ def resolve_metrics(
             ]:
                 # NOTE: 20201026 - JPC - we could use the fact that these metric functions return functions rather
                 # than data to optimize compute in the future
-                resolved_metrics[metric_to_resolve.id] = metric_fn(
-                    **metric_provider_kwargs
-                )
+                try:
+                    resolved_metrics[metric_to_resolve.id] = metric_fn(
+                        **metric_provider_kwargs
+                    )
+                except (
+                    ge_exceptions.MetricComputationError,
+                    ge_exceptions.InvalidMetricAccessorDomainKwargsKeyError,
+                ) as e:
+                    raise ge_exceptions.MetricResolutionError(
+                        message=str(e), failed_metrics=(metric_to_resolve,)
+                    )
             elif metric_fn_type == MetricFunctionTypes.VALUE:
-                resolved_metrics[metric_to_resolve.id] = metric_fn(
-                    **metric_provider_kwargs
-                )
+                try:
+                    resolved_metrics[metric_to_resolve.id] = metric_fn(
+                        **metric_provider_kwargs
+                    )
+                except (
+                    ge_exceptions.MetricComputationError,
+                    ge_exceptions.InvalidMetricAccessorDomainKwargsKeyError,
+                ) as e:
+                    raise ge_exceptions.MetricResolutionError(
+                        message=str(e), failed_metrics=(metric_to_resolve,)
+                    )
             else:
                 logger.warning(
                     f"Unrecognized metric function type while trying to resolve {str(metric_to_resolve.id)}"
                 )
-                resolved_metrics[metric_to_resolve.id] = metric_fn(
-                    **metric_provider_kwargs
-                )
+                try:
+                    resolved_metrics[metric_to_resolve.id] = metric_fn(
+                        **metric_provider_kwargs
+                    )
+                except (
+                    ge_exceptions.MetricComputationError,
+                    ge_exceptions.InvalidMetricAccessorDomainKwargsKeyError,
+                ) as e:
+                    raise ge_exceptions.MetricResolutionError(
+                        message=str(e), failed_metrics=(metric_to_resolve,)
+                    )
         if len(metric_fn_bundle) > 0:
-            resolved_metrics.update(self.resolve_metric_bundle(metric_fn_bundle))
+            try:
+                new_resolved = self.resolve_metric_bundle(metric_fn_bundle)
```

Comment:
Very cool! So if this line fails then we don't update the `resolved_metrics` dict.